backend/dataall/modules/datasets/indexers/dataset_catalog_indexer.py

```diff
@@ -19,8 +22,13 @@ def index(self, session) -> int:
         all_datasets: [Dataset] = DatasetRepository.list_all_active_datasets(session)
         log.info(f'Found {len(all_datasets)} datasets')
         indexed = 0
+        dataset_count = 0
         for dataset in all_datasets:
             tables = DatasetTableIndexer.upsert_all(session, dataset.datasetUri)
             folders = DatasetLocationIndexer.upsert_all(session, dataset_uri=dataset.datasetUri)
-            indexed += len(tables) + len(folders) + 1
+            # Upsert a dataset which doesn't have a table or folder
+            if not DatasetTableRepository.find_all_active_tables(session, dataset.datasetUri) and not DatasetLocationRepository.get_dataset_folders(session, dataset.datasetUri):
```

Comment:
let's just call `DatasetIndexer.upsert(session=session, dataset_uri=dataset.datasetUri)`

no need to add the `if not ...` logic or the additional imports or `dataset_count`

... I understand why but I think we should always just upsert the dataset again (indiscriminate to the dataset having tables / folders or not)

If we see a need to refactor or face some performance issues we can also remove the upsert call for the parent dataset as part of `DatasetTableIndexer` and `DatasetLocationIndexer` at a later date (but this will require some inspection as to where else we call `.upsert(...)` for tables and folders

Reply:
Okay. I see. Will make the changes. It might be repetitive for some datasets. but as you said we can come back and refactor if needed

Reply:
Updated code